### PR TITLE
[FLINK-22177][table-api/doc] Add docs for time functions and time zone support

### PR DESCRIPTION
**Pull request by leonardBang**
## What is the purpose of the change

* This pull request add documents for Table/SQL time functions and time zone support

## Brief change log

  -  Add documents for Table/SQL time functions and time zone support
  - Add examples for  time functions and time zone
  - Add explanation how to support DaylightSavingTime

## Verifying this change

This change is a doc PR without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+## Difference between Batch and Streaming Mode
+The following time functions:
+* LOCALTIME
+* LOCALTIMESTAMP
+* CURRENT_DATE
+* CURRENT_TIME
+* CURRENT_TIMESTAMP
+* NOW()
+
+Flink evaluates their values according to execution mode, i.e. Flink evaluates time function value for row level in Streaming mode, evaluates the time function value at query start for batch mode.
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:309]:
```suggestion
Flink evaluates their values according to execution mode. They are evaluated for each record in streaming mode. But in batch mode, they are evaluated once as the query starts and uses the same result for every row.
```

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                SUM(b)            FROM src
+            GROUP BY TUMBLE(ts_ltz, INTERVAL '0.003' SECOND);
+ -- the TUMBLE_START    column will be close to ts_ltz (represents timestamp value in Asia/Shanghai), the type of TUMBLE_START    column is TIMESTAMP
+ -- the TUMBLE_END      column will be close to ts_ltz + INTERVAL '0.003' SECOND,             the type of TUMBLE_END      column is TIMESTAMP      
+ -- the TUMBLE_ROWTIME  column will be close to ts_ltz + INTERVAL '0.002' SECOND,             the type of TUMBLE_ROWTIME  column is TIMESTAMP_LTZ   
+            
+```
+
+## Daylight Saving Time Support
+Flink SQL supports defining time attributes on TIMESTAMP_LTZ column, base on this, Flink SQL gently uses TIMESTAMP and TIMESTAMP_LTZ type in window processing to support the Daylight Saving Time.
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:283]: gently => gracefully?

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+   
+  
+Flink use timestamp literal to split the window and assigns window to data according to the epoch time of the each row. It means Flink uses TIMESTAMP type for window start and window end(e.g. TUMBLE_START and TUMBLE_END), uses TIMESTAMP_LTZ for window time attribute(e.g. TUMBLE_PROCTIME, TUMBLE_ROWTIME).
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:286]:
```suggestion
Flink use timestamp literal to split the window and assigns window to data according to the epoch time of the each row. It means Flink uses `TIMESTAMP` type for window start and window end (e.g. `TUMBLE_START` and `TUMBLE_END`), uses `TIMESTAMP_LTZ` for window time attribute (e.g. `TUMBLE_PROCTIME`, `TUMBLE_ROWTIME`).
```

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+Giving a example of tumble window, The DaylightTime in Los_Angele start at time 2021-03-14 02:00:00:
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:288]:
```suggestion
Given a example of tumble window, the DaylightTime in Los_Angele starts at time 2021-03-14 02:00:00:
```

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+Please see [Time Attribute]({{< ref "docs/dev/table/concepts/time_attributes">}}#date-and-time) for more information about time attribute.
+
+### Processing Time Attribute and Time Zone
+Flink SQL definine process time attribute by function PROCTIME(), the function return type is TIMESTAMP_LTZ.
+
+**NOTE**
+Before Flink 1.13, the function return type of PROCTIME() is TIMESTAMP, and the return value is the TIMESTAMP in UTC time zone,
+E.g. the PROCTIME() returns `2021-03-01 04:00:00` when user is in Shanghai and current timestamp is `2021-03-01 12:00:00 (UTC+8)`. 
+Flin 1.13 fixes this issue and uses TIMESTAMP_LTZ type as return type of PROCTIME(), users don't need to deal time zone proctime anymore.
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:184]:
```suggestion
{{< hint info >}}
Before Flink 1.13, the function return type of `PROCTIME()` is `TIMESTAMP`, and the return value is the `TIMESTAMP` in UTC time zone,
e.g. the wall-clock shows `2021-03-01 12:00:00` at Shanghai, however the `PROCTIME()` displays `2021-03-01 04:00:00` which is wrong.
Flin 1.13 fixes this issue and uses `TIMESTAMP_LTZ` type as return type of `PROCTIME()`, users don't need to deal time zone problems anymore.
{{< /hint >}}
```

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+                TUMBLE_START(proctime, INTERVAL '0.003' SECOND),
+                TUMBLE_END(proctime, INTERVAL '0.003' SECOND),
+                TUMBLE_PROCTIME(proctime, INTERVAL '0.003' SECOND),
+                COUNT(a),
+                SUM(b)
+            FROM src
+            GROUP BY TUMBLE(proctime, INTERVAL '0.003' SECOND);
+ -- the TUMBLE_START    column will be close to 2021-04-15 22:48:34.567, the type of TUMBLE_START    column is TIMESTAMP
+ -- the TUMBLE_END      column will be close to 2021-04-15 22:48:34.570, the type of TUMBLE_END      column is TIMESTAMP      
+ -- the TUMBLE_PROCTIME column will be close to 2021-04-15 22:48:34.570, the type of TUMBLE_PROCTIME column is TIMESTAMP_LTZ
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:229]: This is hard to understand. Would be better to show some result example here and explain return type of `TUMBLE_START`, `TUMBLE_END`, `TUMBLE_PROCTIME` separately.

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+             'connector' = 'kafka',
+             ...);
+Flink SQL>  SELECT 
+                TUMBLE_START(proctime, INTERVAL '0.003' SECOND),
+                TUMBLE_END(proctime, INTERVAL '0.003' SECOND),
+                TUMBLE_PROCTIME(proctime, INTERVAL '0.003' SECOND),
+                COUNT(a),
+                SUM(b)
+            FROM src
+            GROUP BY TUMBLE(proctime, INTERVAL '0.003' SECOND);
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:226]: Better to use `10` minutes as window interval, which is easier for example.

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+                          ...);
+Flink SQL>  SELECT
+                TUMBLE_START(ts, INTERVAL '0.003' SECOND), 
+                TUMBLE_END(ts, INTERVAL '0.003' SECOND),
+                TUMBLE_ROWTIME(ts, INTERVAL '0.003' SECOND),
+                COUNT(a),
+                SUM(b)
+            FROM src
+            GROUP BY TUMBLE(ts, INTERVAL '0.003' SECOND); 
+ -- the TUMBLE_START    column will be close to ts,                           the type of TUMBLE_START    column is TIMESTAMP
```
Comment (wuchong) [inline docs/content/docs/dev/table/concepts/timezone.md:252]: ditto.

File: docs/content.zh/docs/dev/table/concepts/timezone.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+            FROM src
+            GROUP BY TUMBLE(proctime, INTERVAL '0.003' SECOND);
+ -- the TUMBLE_START    column will be close to 2021-04-15 22:48:34.567, the type of TUMBLE_START    column is TIMESTAMP
+ -- the TUMBLE_END      column will be close to 2021-04-15 22:48:34.570, the type of TUMBLE_END      column is TIMESTAMP      
+ -- the TUMBLE_PROCTIME column will be close to 2021-04-15 22:48:34.570, the type of TUMBLE_PROCTIME column is TIMESTAMP_LTZ      
+          
+```
+
+### Event Time Attribute and Time Zone
+Flink supports defining event time attribute on TIMESTAMP column and TIMESTAMP_LTZ column. It's recommended to defining event time attribute on TIMESTAMP column if the data source contains TIMESTAMP literal.
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/concepts/timezone.md:233]:
```suggestion
Flink supports defining event time attribute on TIMESTAMP column and TIMESTAMP_LTZ column. 
If the timestamp data in the source is represented as year-month-day-hour-minute-second, usually a string value without time-zone information, e.g. `2020-04-15 20:13:40.564`, it's recommended to define the event time attribute as a `TIMESTAMP` column:  
```

File: docs/content.zh/docs/dev/table/concepts/timezone.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+                COUNT(a),
+                SUM(b)
+            FROM src
+            GROUP BY TUMBLE(ts, INTERVAL '0.003' SECOND); 
+ -- the TUMBLE_START    column will be close to ts,                           the type of TUMBLE_START    column is TIMESTAMP
+ -- the TUMBLE_END      column will be close to ts + INTERVAL '0.003' SECOND, the type of TUMBLE_END      column is TIMESTAMP      
+ -- the TUMBLE_ROWTIME  column will be close to ts + INTERVAL '0.002' SECOND, the type of TUMBLE_ROWTIME  column is TIMESTAMP                 
+```
+
+If the data source contains long epoch time, it's recommended to defining event time attribute on TIMESTAMP_LTZ.
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/concepts/timezone.md:256]:
```suggestion
If the timestamp data in the source is represented as a epoch time, usually a long value, e.g. `1618989564564`, it's recommended to define event time attribute as a `TIMESTAMP_LTZ` column.  
```

File: docs/content/docs/dev/table/concepts/timezone.md [hunk 11]
```diff
@@ -0,0 +1,311 @@
+---
+title: "Timestamp DataTypes and Time Zone Support"
```
[the remark this answers is not in the document]
Comment (leonardBang) [reply docs/content/docs/dev/table/concepts/timezone.md:2]: I learned this from other vendors, I'm ok to rename to `Time Zone`

**Comment by leonardBang** [comment]
Thanks @wuchong for the detailed review, I've address your comments and add more example.

**Pull request closed without merging**